src/math/crypto/curve/ec_point.zig

```diff
@@ -0,0 +1,150 @@
+// code ported from starknet-curve:
+// https://github.com/xJonathanLEI/starknet-rs/blob/0857bd6cd3bd34cbb06708f0a185757044171d8d/starknet-curve/src/ec_point.rs
+const Felt252 = @import("../../fields/starknet.zig").Felt252;
+const ALPHA = @import("./curve_params.zig").ALPHA;
+const BETA = @import("./curve_params.zig").BETA;
+
+pub const ProjectivePoint = struct {
+    const Self = @This();
+
+    x: Felt252,
+    y: Felt252,
+    z: Felt252,
+    infinity: bool,
+
+    pub fn from_affine_point(p: AffinePoint) Self {
```

Comment:
The convention for zig functions at the moment is to put camel case in the function names. It's sometimes strange...

![image](https://github.com/keep-starknet-strange/ziggy-starkdust/assets/60488569/097fa51c-218c-4302-a16b-0dc6cf2a5d17)